docker/Dockerfile

```diff
@@ -245,6 +245,10 @@ ENV NAUTOBOT_INSTALLATION_METRICS_ENABLED=false
 RUN nautobot-server init && \
     nautobot-server build_ui
 
+# switch to root user for final-dev stage: https://github.com/nautobot/nautobot/issues/4300
+# hadolint ignore=DL3002
+USER root
+
```

Comment:
Are there any concerns/issues with these commands here: [https://github.com/nautobot/nautobot-plugin-golden-config/blob/develop/development/Dockerfile#L72-L76 being run](https://github.com/nautobot/nautobot-plugin-golden-config/blob/9cc0dbbb5d54c74dce539cc7ae986b1e0dd6b2b6/development/Dockerfile#L72-L76) being run as a different user than the `pip` command installing Nautobot and other packages above?

Reply:
Looks like we'll be fine. We're using `ENV VIRTUAL_ENV="/opt/nautobot"` to tell poetry and pip to install python packages to `/opt/nautobot` so even as root when you run a `pip install xyz` in the `final-dev` image, it installs to `/opt/nautobot`

Reply:
@bryanculver This was something we explicitly tested as well FYI, which was building the `Dockerfile` you linked from the revised `final-dev` container. It was a win.